### PR TITLE
Add custom message

### PR DESCRIPTION
**Pull request by georgecodes**
We want to customise the message that is sent to Github. This allows us to define an external function to do just that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +77,18 @@ Alternatively, click the button below:
...
 
+## Configuring a custom message
+
+If you find the default PR comment a bit dry, it is possible to configure an external message generator which will be called prior to invoking the PR comment. Put a file called message.js in the root of your running mention-bot instance, that looks something like this:
+
+```
+var util = require('util');
+
+module.exports = function(reviewerList, mentionSentenceBuilder) {
+	return util.format("Please review this %s", mentionSentenceBuilder(reviewerList));
```
Comment (vjeux) [inline README.md]: nit: two spaces instead of tab

File: README.md [hunk 2]
```diff
@@ -77,6 +77,18 @@ Alternatively, click the button below:
 
 [![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)
 
+## Configuring a custom message
+
+If you find the default PR comment a bit dry, it is possible to configure an external message generator which will be called prior to invoking the PR comment. Put a file called message.js in the root of your running mention-bot instance, that looks something like this:
+
+```
```
Comment (vjeux) [inline README.md]: nit: ```javascript

File: server.js [hunk 3]
```diff
@@ -86,6 +100,7 @@ app.post('/', function(req, res) {
       };
 
       if (!err && configRes) {
+
```
Comment (vjeux) [inline server.js]: nit: remove

File: server.js [hunk 4]
```diff
@@ -9,11 +9,25 @@
...
 var bl = require('bl');
 var express = require('express');
 var mentionBot = require('./mention-bot.js');
 var GitHubApi = require('github');
 var config = require('./package.json').config;
+var util = require('util');
+var fs = require('fs');
+var configuredMessageGenerator = optional("./message.js");
+var defaultMessageGenerator = function(reviewers) {
+  return util.format("By analyzing the blame information on this pull request" +
```
Comment (vjeux) [inline server.js]: nit: please only indent by two spaces, use single quote for strings and no need to put parenthesis around the conditional anymore:

```javascript
return util.format(
  'By analyzing the blame information on this pull request' + 
  ', we identified %s to be%s potential reviewer%s',
  buildMentionSentence(reviewers),
  reviewers.length > 1 ? '' : ' a', 
  reviewers.length > 1 ? 's' : ''
);
```

File: server.js [hunk 5]
```diff
@@ -9,11 +9,25 @@
...
+var configuredMessageGenerator = optional("./message.js");
+var defaultMessageGenerator = function(reviewers) {
+  return util.format("By analyzing the blame information on this pull request" + 
+                     ", we identified %s to be%s potential reviewer%s",
+                     buildMentionSentence(reviewers),
+                     (reviewers.length > 1 ? '' : ' a'), 
+                     (reviewers.length > 1 ? 's' : ''));
+};
+
+var messageGenerator = configuredMessageGenerator ? configuredMessageGenerator : defaultMessageGenerator;
```
Comment (vjeux) [inline server.js]: You can write that more succinctly

```javascript
var messageGenerator = configuredMessageGenerator || defaultMessageGenerator;
```

File: server.js [hunk 6]
```diff
@@ -9,11 +9,25 @@
 
 require('babel-core/register');
 
+var optional = require("optional");
```
Comment (vjeux) [inline server.js]: I don't really like the optional thing. What about you always have a message.js file where the implementation is something like:

```javascript

module.exports = function(reviewers, defaultMessageGenerator, buildMentionSentence) {

  // If you want to customize the message, put your own custom logic here

  return defaultMessageGenerator(reviewers);
}

File: README.md [hunk 7]
```diff
@@ -77,6 +77,18 @@ Alternatively, click the button below:
...
+var util = require('util');
+
+module.exports = function(reviewerList, mentionSentenceBuilder) {
```
Comment (vjeux) [inline README.md]: put this in comment in the mention.js file.

**Comment by vjeux** [comment]
Thanks for the pull request, made a lot of comments, but once you address them it'll be glorious :)

**Comment by georgecodes** [comment]
Comments welcome, thanks. Will address them shortly.

**Comment by jgardella** [comment]
This looks good; it's different and more powerful than the approach I was considering. I was thinking of adding an option to the repo-config which would be a list of messages, and then the bot would randomly pick from among those messages. We could possibly implement both and use the config messages as a fallback if message.js isn't defined. The advantage of defining it in the config is that you wouldn't have to roll out your own version of the bot to add new messages.

File: server.js (hunk 5 above)
Comment (georgecodes) [inline server.js]: I *knew* there was a more JS-y way to do that. I was stuck in Groovy mode and trying to do an elvis operator

File: README.md (hunk 7 above)
Comment (georgecodes) [inline README.md]: mention.js?

**Comment by vjeux** [comment]
Thanks!

**Pull request merged**